docs/strain.schema.json

```diff
@@ -0,0 +1,66 @@
+{
+    "meta:license": [
+        "Copyright 2019 Adobe. All rights reserved.",
+        "This file is licensed to you under the Apache License, Version 2.0 (the \"License\");",
+        "you may not use this file except in compliance with the License. You may obtain a copy",
+        "of the License at http://www.apache.org/licenses/LICENSE-2.0",
+        "",
+        "Unless required by applicable law or agreed to in writing, software distributed under",
+        "the License is distributed on an \"AS IS\" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS",
+        "OF ANY KIND, either express or implied. See the License for the specific language",
+        "governing permissions and limitations under the License."
+    ],
+    "$id": "https://ns.adobe.com/helix/shared/strain",
+    "$schema": "http://json-schema.org/draft-07/schema#",
+    "title": "Strain",
+    "type": "object",
+    "meta:status": "stabilizing",
+    "description": "A strain is a combination of code and content that enables the creation of a digital experience. Strains can be used to create language variants of websites, A/B tests, personalization, or to aggregate content from multiple sources",
+    "additionalProperties": false,
+    "properties": {
+        "code": {
+            "description": "Pointer to the code repository",
+            "oneOf": [
+                {
+                    "type": "string",
+                    "format": "uri"
+                },
+                {
+                    "$ref": "https://ns.adobe.com/helix/shared/giturl"
+                }
+            ]
+        },
+        "content": {
+            "description": "Pointer to the content repository",
+            "oneOf": [
+                {
+                    "type": "string",
+                    "format": "uri"
+                },
+                {
+                    "$ref": "https://ns.adobe.com/helix/shared/giturl"
+                }
+            ]
+        },
+        "static": {
+            "description": "Pointer to the repository for static resources",
+            "oneOf": [
+                {
+                    "type": "string",
+                    "format": "uri"
+                },
+                {
+                    "$ref": "https://ns.adobe.com/helix/shared/giturl"
+                }
+            ]
+        },
+        "sticky": {
+            "description": "Sticky strains are not re-evaluated on every request. As soon as a visitor is determined to match a sticky strain, a session cookie will be set to keep the user in the strain.",
+            "type": "boolean"
+        }
+    },
+    "required": [
+        "code",
+        "content"
```

Comment:
why isn't `static` required, too? default to `code` ?

sidenote:
if `code` and `static` point to the same repo, shouldn't `hlx deploy` also fixate the ref of the `static` repo with the deployment hash?